Docs/source/developers/python.rst

```diff
@@ -1,7 +1,87 @@
 .. _development-python:
 
-Python interface
-================
+Handling PICMI input files
+==========================
+
+There are two levels in the handling of the input files. The first is the Python level API that mirrors the input parameters and the
+second is the translation from the PICMI input to the input parameters. The two levels are described below.
+
+Input parameters
+----------------
+
+In a C++ input file, each of the parameters has a prefix, for example ``geometry`` in ``geometry.prob_lo``.
+For each of these prefixes, an instance of a Python class is created and the parameters saved as attributes.
+This construction is used since the lines in the input file look very much like a Python assignment statement,
+assigning attributes of class instances, for example ``geometry.dims = 3``.
+
+Many of the prefix instances are predefined, for instance ``geometry`` is created in the file ``Python/pywarpx/Geometry.py``.
+In that case, ``geometry`` is an instance of the class ``Bucket`` (specified in ``Python/pywarpx/Bucket.py``),
+the general class for prefixes.
+It is called ``Bucket`` since its main purpose is a place to hold attributes.
+Most of the instances are instances of the ``Bucket`` class.
+There are exceptions, such as ``constants`` and ``diagnostics`` where extra processing is needed.
+
+There can also be instances created as needed.
+For example, for the particle species, an instance is created for each species listed in ``particles.species_names``.
+This gives a place to hold the parameters for the species, for instance ``electrons.mass``.
```

Comment:
To avoid confusion with all the *instance(s)*:
```suggestion
This gives a place to hold the parameters for the species, e.g., ``electrons.mass``.
```